app/services/facebook_authenticator.rb

```diff
@@ -3,79 +3,47 @@
 # Handles Facebook authentication
 #
 # @!visibility private
-class FacebookAuthenticator
+class FacebookAuthenticator < BaseAuthenticator
 
-  class ApiError < StandardError; end
-
-  def initialize(auth_code)
-    @auth_code = auth_code
-  end
-
-  def authenticate!
-    facebook_user = get_facebook_user(access_token)
-    login         = find_login(facebook_user)
-
-    if login.present?
-      connect_login_to_fb_account(login, facebook_user)
-    else
-      login = create_login_from_fb_account(facebook_user)
-    end
-
-    login
-  end
+  PROVIDER = 'facebook'
+  TOKEN_URL = 'https://graph.facebook.com/oauth/access_token?client_id=%{client_id}&client_secret=%{client_secret}&code=%{auth_code}&redirect_uri=%{redirect_uri}'
+  PROFILE_URL = 'https://graph.facebook.com/me?fields=email,name&access_token=%{access_token}'
```

Comment:
The graph URL should be read from the config (`RailsApiAuth.facebook_graph_url`).